encryptedlogging/src/test/kotlin/com/automattic/encryptedlogging/persistence/dao/EncryptedLogDaoTest.kt

```diff
@@ -0,0 +1,236 @@
+        assertThat(sut.getEncryptedLog(TEST_UUID))
+            .usingRecursiveComparison()
+            .ignoringFields("id")
+            .isEqualTo(initialLog)
+
+        // Get the encrypted log from the database (to get the correct id)
+        val initialLogFromDb = sut.getEncryptedLog(TEST_UUID)
+        assertThat(initialLogFromDb).isNotNull
+
+        // Create a copy of the encrypted log by changing its upload state (which will be the common usage)
+        val newUploadState = EncryptedLogUploadState.UPLOADING
+        val updatedLog = initialLogFromDb?.copy(uploadStateDbValue = newUploadState.value)
+        updatedLog?.let { sut.upsertEncryptedLog(updatedLog) }
+
+        // Assert that the encrypted log in the DB is the one with the correct upload state
+        val updatedLogFromDB = sut.getEncryptedLog(TEST_UUID)
+        assertThat(requireNotNull(updatedLogFromDB?.uploadState)).isEqualTo(newUploadState)
+        // This verifies the expected state as well but separating the initial assertion is valuable to show intent
+        assertThat(updatedLogFromDB).isEqualTo(updatedLog)
+    }
+
+    @Test
+    fun `test delete encrypted log`() = runTest {
+        // Insert an initial encrypted log
+        val initialLog = createTestEncryptedLogModel()
+        sut.insertEncryptedLog(initialLog)
+        assertThat(sut.getEncryptedLog(TEST_UUID))
+            .usingRecursiveComparison()
+            .ignoringFields("id")
+            .isEqualTo(initialLog)
+
+        // Get the encrypted log from the database (to get the correct id)
+        val initialLogFromDb = sut.getEncryptedLog(TEST_UUID)
+        assertThat(initialLogFromDb).isNotNull
+
+        // Delete the encrypted log
+        initialLogFromDb?.let { sut.deleteEncryptedLog(it) }
+
+        // Assert that the encrypted log no longer exists
+        assertThat(sut.getEncryptedLog(TEST_UUID)).isNull()
+    }
+
+    @Test
+    fun `test get uploading encrypted logs`() = runTest {
+        // Insert an encrypted log with uuid
+        val logToBeInserted = createTestEncryptedLogModel(uploadState = EncryptedLogUploadState.UPLOADING)
+        sut.insertEncryptedLog(logToBeInserted)
+
+        // Assert that the encrypted log from the DB is the same as the one we inserted (ignoring id)
+        val uploadingEncryptedLogs = sut.getEncryptedLogs(
+            EncryptedLogUploadState.UPLOADING.value
+        )
+        assertThat(uploadingEncryptedLogs).hasSize(1)
+        assertThat(uploadingEncryptedLogs.first())
+            .usingRecursiveComparison()
+            .ignoringFields("id")
+            .isEqualTo(logToBeInserted)
+    }
+
+    @Test
+    fun `test uploading encrypted logs count for empty DB`() = runTest {
+        assertThat(sut.getEncryptedLogsCount(EncryptedLogUploadState.UPLOADING.value)).isEqualTo(0)
+    }
+
+    @Test
+    fun `test uploading encrypted logs for random number`() = runTest {
+        Random.nextInt(100).let { numberOfLogs ->
```

Comment:
We should not use random in tests. I don't think this brings any value, comparing to a predefined constant, while introducing randomness which is difficult to debug.

Reply:
PS: [Similar](https://github.com/Automattic/EncryptedLogging/pull/20#discussion_r2164016781)